src/Server/HTTPHandler.cpp

```diff
@@ -94,12 +104,31 @@ namespace ErrorCodes
     extern const int UNKNOWN_USER;
     extern const int WRONG_PASSWORD;
     extern const int REQUIRED_PASSWORD;
+    extern const int AUTHENTICATION_FAILED;
 
     extern const int BAD_REQUEST_PARAMETER;
     extern const int INVALID_SESSION_TIMEOUT;
     extern const int HTTP_LENGTH_REQUIRED;
 }
 
+static String base64Decode(const String & encoded)
+{
+    String decoded;
+    Poco::MemoryInputStream istr(encoded.data(), encoded.size());
+    Poco::Base64Decoder decoder(istr);
+    Poco::StreamCopier::copyToString(decoder, decoded);
+    return decoded;
+}
+
+static String base64Encode(const String & decoded)
+{
+    std::ostringstream ostr; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
```

Comment:
```suggestion
    std::ostringstream ostr; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
    ostr.exceptions(std::ios::failbit);
```
see https://github.com/ClickHouse/ClickHouse/pull/16772

Reply:
`Poco::Base64Encoder` accepts only `std::ostream`, and `WriteBufferFromOwnString` doesn't implement that interface.